spectrocloud/cluster_common_fields.go

```diff
@@ -135,7 +136,7 @@ func getSpectroComponentsUpgrade(cluster *models.V1SpectroCluster) string {
 
 // update common fields like namespaces, cluster_rbac_binding, cluster_profile, backup_policy, scan_policy
 func updateCommonFields(d *schema.ResourceData, c *client.V1Client) (diag.Diagnostics, bool) {
-	if d.HasChanges("name", "tags", "description") {
+	if d.HasChanges("name", "tags", "tags_map", "description") {
```

Comment:
@stefanSpectro We need to handle this differently. The issue is that the `tags_map` attribute is only available for `AWS` and `EKS` clusters. For other cluster types, this attribute is not present in the schema, which leads to a plugin crash if it's accessed or set.

To avoid this, we should ensure tags_map is only referenced conditionally or as part of schemas where it's explicitly supported.